### PR TITLE
Jettae.schroff

### PR DESCRIPTION
**Pull request by Parzing**
Completed ScanningCensorshipService

### CODE REVIEW

**Comment by tjcaul** [comment]
It's accidentally adding a space after the end, so the strings don't match 💀

File: newspeek/src/main/java/use_case/helpers/ScanningCensorshipService.java [hunk 1]
```diff
@@ -4,23 +4,33 @@
 import entity.censorship_rule_set.CensorshipRuleSet;
 
 public class ScanningCensorshipService implements CensorshipService {
+
+    private static final String WORD_REGEX = "[ -.,()\\[\\]{};:\"']+";
```
Comment (NotAName320) [inline newspeek/src/main/java/use_case/helpers/ScanningCensorshipService.java:8]: could be nice to add comments for regex strings

**Review by NotAName320 — CHANGES_REQUESTED** [review]

File: newspeek/src/main/java/use_case/helpers/ScanningCensorshipService.java [hunk 2]
```diff
@@ -17,24 +17,77 @@ public Article censor(Article article, CensorshipRuleSet ruleset) {
...
+            result.setText(resultText.toString());
+            return true;
+        }
+        else if (punctuation.length == 0){
+            resultText.append(words[0]);
+            result.setText(resultText.toString());
+            return true;
+        }
+        else return words[0].isEmpty() && punctuation[0].isEmpty();
+    }
```
Comment (NotAName320) [inline newspeek/src/main/java/use_case/helpers/ScanningCensorshipService.java:83]: since each branch returns we don't need the elses

**Review by NotAName320 — CHANGES_REQUESTED** [review]

**Review by tjcaul — APPROVED** [review]
Looks like you've copied and adapted an existing test to test replacement as well. It tests both case-sensitive and case-insensitive, and it passes! noice 👍

File: newspeek/src/main/java/use_case/helpers/ScanningCensorshipService.java (hunk 2 above)
Comment (tjcaul) [reply newspeek/src/main/java/use_case/helpers/ScanningCensorshipService.java:83]: This is highly debatable

**Review by NotAName320 — APPROVED** [review]

**Pull request merged**